File: lab2/app/src/main/java/com/example/hellokitty_java/MainActivity.java
package com.example.hellokitty_java;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText editText;
    private TextView mHelloTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        mHelloTextView = findViewById(R.id.textView);
        editText = findViewById(R.id.editText);
        ImageButton imageButton = findViewById(R.id.imageButton);

        imageButton.setOnClickListener(v -> {
            if (editText.getText().toString().isEmpty()) {
                mHelloTextView.setText("Hello Kitty!");
            } else {
                mHelloTextView.setText("Привет, " + editText.getText().toString());
            }
        });

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: lab3/app/src/main/java/com/example/ravencounter/MainActivity.kt
package com.example.ravencounter

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private var raven_counter: Int=0;
    private var cat_counter: Int = 0;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun onClickCat(view: View) {
        val textViewCat: TextView= findViewById(R.id.textView_cat)
        textViewCat.setText("Я насчитала ${++cat_counter} котиков")
    }

    fun onClickRaven(view: View) {
        val textViewRaven: TextView= findViewById(R.id.textView_raven)
        textViewRaven.setText("Я насчитала ${++raven_counter} ворон")
    }


}